contracts/wrapper/NameWrapper.sol

```diff
@@ -621,18 +639,19 @@ contract NameWrapper is
     }
 
     function _unwrap(bytes32 node, address newOwner) private {
-        require(
-            newOwner != address(0x0),
-            "NameWrapper: Target owner cannot be 0x0"
-        );
-        require(
-            newOwner != address(this),
-            "NameWrapper: Target owner cannot be the NameWrapper contract"
-        );
-        require(
-            !allFusesBurned(node, CANNOT_UNWRAP),
-            "NameWrapper: Domain is not unwrappable"
-        );
+        if (newOwner == address(0x0)) {
+            // Target owner cannot be the 0x0 contract
+            revert IncorrectTargetOwner(newOwner);
+        }
+
+        if (newOwner == address(this)) {
+            // Target owner cannot be the NameWrapper contract
+            revert IncorrectTargetOwner(newOwner);
+        }
+
+        if (allFusesBurned(node, CANNOT_UNWRAP)) {
+            revert OperationProhibited(node);
```

Comment:
Perhaps we should log the fuse it was checking against too?

Reply:
I thought about this. It's easy here to log CANNOT_UNWRAP, but less obvious for this:

(owner == address(0) && fuses & CANNOT_CREATE_SUBDOMAIN != 0) ||
	                (owner != address(0) && fuses & CANNOT_REPLACE_SUBDOMAIN != 0)

Reply:
```
        if((owner == address(0) && fuses & CANNOT_CREATE_SUBDOMAIN != 0)){
            OperationProhibited(node, CANNOT_CREATE_SUBDOMAIN);
        }

        if((owner != address(0) && fuses & CANNOT_REPLACE_SUBDOMAIN != 0)){
            OperationProhibited(node, CANNOT_REPLACE_SUBDOMAIN);
        }

```

It's not quite right, since it's also dependent on whether the owner is 0 or not...

Reply:
Also, i'm trying to add `OperationProhibited(node, fuses)` to `ERC1155Fuse`, but it doesn't inherit from INameWrapper which has all the constants for fuses. What's the best way to get the CAN_TRANSFER fuse?

We call `_canTransfer` inside this contract instead of using the fuse directly so I don't have access it (bar hardcoding the number)

Reply:
Good point. Let's leave it as-is.